snapcraft/plugins/dump.py

```diff
@@ -35,7 +35,58 @@ class DumpPlugin(snapcraft.BasePlugin):
 
     def build(self):
         super().build()
-        if os.path.exists(self.installdir):
-            os.rmdir(self.installdir)
-        shutil.copytree(self.builddir, self.installdir,
-                        copy_function=snapcraft.common.link_or_copy)
+        _linktree(self.builddir, self.installdir, self.installdir)
+
+
+def _link_or_copy(source, destination, boundary):
```

Comment:
Yeah good point. Though I did it in the order of calls, which flipped the last two.